python/pyspark/sql/connect/client.py

```diff
@@ -551,37 +588,35 @@ def _execute_and_fetch(
         logger.info("ExecuteAndFetch")
 
         m: Optional[pb2.ExecutePlanResponse.Metrics] = None
-
         batches: List[pa.RecordBatch] = []
 
         try:
-            for b in self._stub.ExecutePlan(req, metadata=self._builder.metadata()):
-                if b.client_id != self._session_id:
-                    raise SparkConnectException(
-                        "Received incorrect session identifier for request."
-                    )
-                if b.metrics is not None:
-                    logger.debug("Received metric batch.")
-                    m = b.metrics
-                if b.HasField("arrow_batch"):
-                    logger.debug(
-                        f"Received arrow batch rows={b.arrow_batch.row_count} "
-                        f"size={len(b.arrow_batch.data)}"
-                    )
-
-                    with pa.ipc.open_stream(b.arrow_batch.data) as reader:
-                        for batch in reader:
-                            assert isinstance(batch, pa.RecordBatch)
-                            batches.append(batch)
+            for attempt in Retrying(SparkConnectClient.retry_exception, **self._retry_policy):
+                with attempt:
```

Comment:
Hm ... I feel like we should do something with this `attempt` and the actual request..

Reply:
Otherwise, we might just need one `retry` context manager for `with` statement instead of the combination w/ ganarator and `with` statement together.

Reply:
I think I'm confused, as explained above, you need the context manager to capture the exception that is thrown, but the contextmanager cannot retry itself (throws a do not repeat this exception) this is why you need the generator.

The alternative is to wrap all of the API stubs into much more ugly code and I personally find this rather idiomatic because the actual exception is still visible and not swallowed.